pkg/analyzer/analyzer.go

```diff
@@ -82,38 +255,53 @@ func checkArgument(pass *analysis.Pass, v ast.Expr) *analysis.Diagnostic {
 	}
 }
 
+func checkDurationConversionArgument(pass *analysis.Pass, arg ast.Expr) *analysis.Diagnostic {
+	argType := pass.TypesInfo.TypeOf(arg)
+	implicitInt := usesIntOrUntypedConstants(pass.TypesInfo, arg)
+	explicitInt := isIntegerType(argType)
+	if !implicitInt && !explicitInt {
+		return nil
+	}
+	return &analysis.Diagnostic{
+		Pos:     arg.Pos(),
+		Message: "converting integer via time.Duration() without multiplication by proper duration",
+	}
+}
+
 func checkAssignment(pass *analysis.Pass, l ast.Expr, r ast.Expr) *analysis.Diagnostic {
 	lType := pass.TypesInfo.TypeOf(l)
 	if lType == nil || lType.String() != "time.Duration" {
 		return nil
 	}
-	if !usesUntypedConstants(pass.TypesInfo, r) {
+	if !usesIntOrUntypedConstants(pass.TypesInfo, r) {
 		return nil
 	}
 	return &analysis.Diagnostic{Pos: r.Pos(), Message: "untyped constant in time.Duration assignment"}
 }
 
-func usesUntypedConstants(ti *types.Info, e ast.Expr) bool {
+func usesIntOrUntypedConstants(ti *types.Info, e ast.Expr) bool {
 	switch v := e.(type) {
 	case *ast.BasicLit: // ex: 1
 		return v.Value != "0"
 	case *ast.BinaryExpr:
 		switch v.Op {
-		case token.ADD: // ex: 1 + time.Seconds
-			return usesUntypedConstants(ti, v.X) || usesUntypedConstants(ti, v.Y)
+		case token.ADD, token.SUB: // ex: 1 + time.Seconds
+			return usesIntOrUntypedConstants(ti, v.X) || usesIntOrUntypedConstants(ti, v.Y)
 		case token.MUL: // ex: 1 * time.Seconds
-			return usesUntypedConstants(ti, v.X) && usesUntypedConstants(ti, v.Y)
+			return usesIntOrUntypedConstants(ti, v.X) && usesIntOrUntypedConstants(ti, v.Y)
 		}
 	case *ast.Ident: // ex: someIdentifier
-		return hasUntypedConstDeclaration(ti, v)
+		return hasIntOrUntypedConstDeclaration(ti, v)
+	case *ast.ParenExpr:
+		return usesIntOrUntypedConstants(ti, v.X)
 	}
 	return false
 }
 
 // we only care about untyped `const Name = 123` declarations
 // `var Name = 123`, and `a := 123` declarations are already type-checked
 // by the compiler
-func hasUntypedConstDeclaration(ti *types.Info, identifier *ast.Ident) bool {
+func hasIntOrUntypedConstDeclaration(ti *types.Info, identifier *ast.Ident) bool {
```

Comment:
This is necessary to handle `Sleep(time.Duration(someExistingIntegerInSecondsConstant))` correctly, but it needs to be rethought to handle a greater variety of cases.